internal/datacoord/compaction.go

```diff
@@ -269,7 +261,23 @@ func (c *compactionPlanHandler) handleMergeCompactionResult(plan *datapb.Compact
 	log.Info("handleCompactionResult: altering metastore after compaction")
 	if err := c.meta.alterMetaStoreAfterCompaction(modInfos, newSegment.SegmentInfo); err != nil {
```

Comment:
merge alterInMemoryMetaAfterCompaction into alterMetaStoreAfterCompaction because I want it to be atomic operation